### ISSUE TITLE
Errors when using TypeScript

### ISSUE THREAD

**Issue by santteegt** [3box/3box-profile-edit-react#8]
Hi,

When integrating the plugin on a dApp using TypeScript, I'm getting the following errors:

```
ERROR in ./node_modules/3box-profile-edit-react/index.js
Module not found: Error: Can't resolve './dist/index' in '/Users/santteegt/GitRepositories/widget-connect/node_modules/3box-profile-edit-react'
 @ ./node_modules/3box-profile-edit-react/index.js 3:17-40
 @ ./src/core/Core.tsx
 @ ./src/core/index.tsx
 @ ./src/index.ts

ERROR in ./node_modules/3box-profile-edit-react/index.js
Module not found: Error: Can't resolve './dist/manifest' in '/Users/santteegt/GitRepositories/widget-connect/node_modules/3box-profile-edit-react'
 @ ./node_modules/3box-profile-edit-react/index.js 1:0-26
 @ ./src/core/Core.tsx
 @ ./src/core/index.tsx
 @ ./src/index.ts

ERROR in ./node_modules/3box-profile-edit-react/index.js
Module not found: Error: Can't resolve './dist/vendor' in '/Users/santteegt/GitRepositories/widget-connect/node_modules/3box-profile-edit-react'
 @ ./node_modules/3box-profile-edit-react/index.js 2:0-24
 @ ./src/core/Core.tsx
 @ ./src/core/index.tsx
 @ ./src/index.ts

ERROR in /Users/santteegt/GitRepositories/widget-connect/src/core/Core.tsx
./src/core/Core.tsx
[tsl] ERROR in /Users/santteegt/GitRepositories/widget-connect/src/core/Core.tsx(489,13)
      TS2769: No overload matches this call.
  Overload 2 of 2, '(props: EditProfileProps, context?: any): EditProfile', gave the following error.
    Type '({ inputType: string; key: string; field: string; options?: undefined; } | { inputType: string; key: string; field: string; options: { value: string; display: string; }[]; })[]' is not assignable to type '[any][]'.
  Overload 2 of 2, '(props: EditProfileProps, context?: any): EditProfile', gave the following error.
    Type '(address: string) => Promise<void>' is not assignable to type '(url: string) => string'.
```

**Comment by santteegt** [comment]
Most of the errors specified above have been solved in `v0.0.1-beta.3`. However, the following component properties aren't well defined on `index.d.ts`:

* `redirectFn` definition differs from how it is defined in the docs: `The user's ethereum address will be passed as a param in the event that the dapp uses this in the route`
https://github.com/3box/3box-profile-edit-react/blob/ebc148cac81e19fc7802c5e1c388b99c534d35a3/index.d.ts#L12

* `customFields` is mandatory instead of optional
https://github.com/3box/3box-profile-edit-react/blob/ebc148cac81e19fc7802c5e1c388b99c534d35a3/index.d.ts#L14

**Comment by oznekenzo** [comment]
@santteegt Fixed!
